src/psyclone/parse.py

```diff
@@ -444,10 +431,15 @@ def __init__(self, api=""):
             self._type = api
 
     def create(self, ast, name=None):
+        '''
+        Create a Kernel object for the API supplied to the constructor
+        of this factory.
 
-        if self._type == "gunghoproto":
-            return GHProtoKernelType(ast, name=name)
-        elif self._type == "dynamo0.1":
+        :param ast: The fparser1 AST for the Kernel code.
+        :type ast: :py:class:`fparser.one.block_statements.BeginSource`
+        :param str name: the name of the kernel or None.
+        '''
+        if self._type == "dynamo0.1":
```

Comment:
In general we try and fix such errors since, apart from at the very top level, we should be catching specific exceptions. However, these aren't in any code I've touched in this issue. I've updated the copyright date.